iStore/Login/LoginView.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  LoginView.swift
+//  iStore
+//
+//  Created by Mohamed Sayed on 13.03.24.
+//
+
+import SwiftUI
+
+struct LoginView: View {
+    
+    @EnvironmentObject var authenticationViewModel: AuthenticationViewModel
+    
+    @State private var userName: String
+    @State private var password: String
+    
+    init(userName: String, password: String) {
+        self.userName = userName
+        self.password = password
+    }
+    
+    var body: some View {
+        VStack(alignment: .leading, spacing: 24) {
+            Text("iStore")
+                .font(.system(.largeTitle, design: .rounded))
+                .fontWeight(.bold)
+                .foregroundColor(.black)
+                .padding(.top, 25)
+                .padding(.horizontal)
+            
+            Text("Get Everything at Your Doorstep")
+                .font(.system(.largeTitle, design: .rounded))
+                .fontWeight(.bold)
+                .foregroundColor(.pink)
+                .padding()
+            
+            Spacer()
+            
+            VStack(spacing: 24) {
+                UserNameStack(userName: $userName)
+                
+                PasswordStack(password: $password)
+                
+                if authenticationViewModel.showError {
+                    Text("Please, Enter a valid UserName and Password!")
+                        .foregroundColor(.red)
+                }
+                
+                if authenticationViewModel.showProgress {
```

Comment:
This is should be `else if`, right? authenticationViewModel can't show error and progress in the same time

Reply:
I think both are valid, but else if is better.
It wouldn't show both at the same time, as I have already handled Its logic from the AuthenticathionViewModel.

Reply:
Its related to the compiler process. Two if conditions means that the compiler needs to check both statements. If else not required to check both statements for all cases